evap/contributor/forms.py

```diff
@@ -66,8 +69,11 @@ def clean_general_questionnaires(self):
         # Ensure all locked questionnaires still have the same status (included or not)
         locked_qs = self.fields['general_questionnaires'].queryset.filter(is_locked=True)
 
-        not_locked = [q for q in self.cleaned_data.get('general_questionnaires') if q not in locked_qs]
-        locked = [q.pk for q in self.instance.general_contribution.questionnaires.filter(is_locked=True)]
+        not_locked = []
+        if self.cleaned_data.get('general_questionnaires'):
+            not_locked = [q for q in self.cleaned_data.get('general_questionnaires') if q not in locked_qs]
```

Comment:
We now know that locked questionnaires as disabled fields aren’t submitted, and I think explicitly filtering out locked questionnaires is a good practise in the sense of defensive programming. But, I think a comment here stating that locked questionnaires aren’t submitted in a normal case would help future people.

Also, the expression could be:
```suggestion
            not_locked = list(self.cleaned_data.get('general_questionnaires').filter(is_locked=False))
```

Reply:
i honestly don't know where to put what comment exactly so that it doesn't feel out of place.

the expression was changed.

Reply:
Well, then no reason to force it :)